### DISCUSSION TITLE
Appwrite Storage Large File Support

### DISCUSSION THREAD

**Discussion by lohanidamodar** [appwrite/appwrite#1283] [Ideas]
Hey Appwriters,
We are looking into efficient ways to handle large file support in Appwrite Storage and we have few concerns regarding large files around anti virus scanning, compression and encryption. We are thinking of skipping virus scanning, compression and encryption for large files above certain size say for example 30 MB. 
We want to open this discussion to get feedback on this approach.
- What approach would you suggest for large file support?
- Do you have any use case that needs encryption and compression for large files?

Let us know your thoughts and suggestions.
Thank You

**Comment by Meldiron** [comment]
I have been using MinIO lately and we were able to successfully upload&store video-lesson archive in there (mp4). MinIO also has great security features (by signing URLs) but that is a topic for another discussion probably.

Usecases I can see from my last experience:
- Video in general (archive, product presentation, animated video backgrounds)
- 3D models (mobile applications such as apartmant preview or product preview in VR)
- source images may be 10+mb if you really need to have source stored (and then resize with something like imgproxy)

How long would virus scanning, compression and encryption take? When uploading 30+ MB file, you always have some progression animation on the frontend, or you even do it on the backend after the video is uploaded and notify user - Youtube is a great example.

**Reply by Meldiron** [reply]
I just noticed all of my use-cases are uploaded by administrator so yeah, virus scanning could be disabled.
The only one problematic one would be "YouTube clone".

**Comment by alexweininger** [comment]
Tracking here: https://github.com/utopia-php/storage/pull/16
